src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -243,20 +223,15 @@ private bool MoreChars()
         /// 
         /// Comments and string literals are allowed to contain any Unicode character.
         /// </summary>
-        /// <returns></returns>
-        internal bool IsReallyAtEnd()
-        {
-            return _offset >= _characterWindowCount && Position >= _textEnd;
-        }
+        public readonly bool IsReallyAtEnd()
+            => Position >= _textEnd;
 
         /// <summary>
-        /// Advance the current position by one. No guarantee that this
-        /// position is valid.
+        /// Advance the current position by one. No guarantee that this position is valid.  This will <em>not</em> change the character window
+        /// in any way.  Specifically, it will not create a new character window, nor will it change the contents of the current window.
```

Comment:
Should `TryAdvance` also get this note about not creating a new window?

Reply:
I have updated the docs.  In actuality TryAdvance *can* change the window.  that's because it needs to *peek* to see the next char, and peeking when on the boundary will bring in the next chunk.